rust/agama-lib/src/questions/http_client.rs

```diff
@@ -89,30 +89,56 @@ impl HTTPClient {
         }
     }
 
-    pub async fn delete_question(&self, question_id: u32) -> Result<(), QuestionsHTTPClientError> {
-        let path = format!("/questions/{}", question_id);
-        Ok(self.client.delete_void(path.as_str()).await?)
-    }
+    pub async fn set_mode(&self, policy: Policy) -> Result<(), QuestionsHTTPClientError> {
+        let questions = QuestionsConfig {
+            policy: Some(policy),
+            ..Default::default()
+        };
+        let config = Config {
+            questions: Some(questions),
+            ..Default::default()
+        };
+
+        let patch = Patch {
+            update: Some(config),
+        };
 
-    pub async fn get_config(&self) -> Result<QuestionsConfig, QuestionsHTTPClientError> {
-        Ok(QuestionsConfig::default())
+        _ = self.client.patch_void("/v2/config", &patch).await?;
+        Ok(())
     }
 
-    pub async fn set_config(
+    pub async fn set_answers(
         &self,
-        config: &QuestionsConfig,
+        answers: Vec<AnswerRule>,
     ) -> Result<(), QuestionsHTTPClientError> {
-        Ok(self.client.put_void("/questions/config", config).await?)
+        let questions = QuestionsConfig {
+            answers,
+            ..Default::default()
+        };
+        let config = Config {
+            questions: Some(questions),
+            ..Default::default()
+        };
+
+        let patch = Patch {
+            update: Some(config),
+        };
+        self.client.patch_void("/v2/config", &patch).await?;
+        Ok(())
+    }
+
+    pub async fn delete_question(&self, id: u32) -> Result<(), QuestionsHTTPClientError> {
+        let update = UpdateQuestion::Delete { id };
+        self.client.patch_void("/v2/questions", &update).await?;
```

Comment:
I hope that v2 is just temporary as I do not see point of having killed v1 and only having v2.

Reply:
Yes, it is a temporary thingie.